app/controllers/concerns/basic_authenticator.rb

```diff
@@ -2,19 +2,52 @@
 
 module BasicAuthenticator
```

Comment:
Per Jonah: Consider moving this into the authenticator controller directly

Reply:
@jonahx, this is currently used by both the `CredentialsController` and the `AuthenticationController`, so I'm inclined to leave it as a concern. Let me know if you have a better design idea though!